feat(profile): multiple camera support + useVideoPreview hook

germanocaumo opened the pull request:
### What does this PR do?
This PR has two changes:

- Video Preview Refactor: Adds a useVideoPreview hook with reusable state and functions of video preview related functions so that they can be reused in profile view and camera as content as well.
    - Centralize core video preview code in one place
    - Convert the code to typescript
    - Remove duplicated code related to camera preview in the profile sidebar and share camera as content
    - The code is still all the same, only re-arranged and converted to function component

- Implements Multiple Camera support in profile settings according to screens from design:
  - the team is still working on this and there could be more changes still
<img width="426" height="892" alt="image" src="https://github.com/user-attachments/assets/9849b978-f6c6-41c1-92cc-703a7fa4d7ea" />

![multipleCamera](https://github.com/user-attachments/assets/a72335be-266e-4d52-bd10-1ee1704f73e8)

### Closes Issue(s)

Closes https://github.com/bigbluebutton/bigbluebutton/issues/23278

### Motivation
Allow to configure two or more cameras via the profile in the sidebar.

germanocaumo committed:
feat(profile): add multiple camera support +

- Implements multiple camera support in profile settings according to screens from design.
- Fixes some bugs in useVideoPreviewHook, mainly the getCameraStream being called multiple times wrongly and skipVideoPreview not showing the preview
when errors occurs (no permission for example).

germanocaumo committed:
chore: missing new locale

germanocaumo committed:
refactor(video-preview): add a reusable useVideoPreview hook

Adds a new hook with reusable state and functions of video preview related functions so that they can be reused
in profile view and camera as content.
- Centralize core video preview code in one place
- Convert the code to typescript
- Remove duplicated code related to camera preview in the profile sidebar and share camera as content

germanocaumo committed:
chore: refactor to fix sonarcloud warnings

germanocaumo committed:
refactor: extract to function to fix sonarcloud warning

gustavotrott commented:
Taking a look at the images, I had some thoughts, just sharing in case you consider it relevant.

---

Is this section always visible? Could it be hidden when only one webcam is detected? 
<img width="453" height="180" alt="image" src="https://github.com/user-attachments/assets/b7ee228f-7202-4343-b583-edd4ff8bc7b3" />

The panel is already quite busy, so adding more static items might make it harder to use, especially on small screens. Would it be possible to place the new button next to *Share webcam*? Maybe a dropdown like this:
<img width="341" height="172" alt="image" src="https://github.com/user-attachments/assets/c3d3a7f3-e908-4451-9961-6a5ed80c2a82" />

---
The *Stop sharing* button often requires to scroll a lot to find it. 
Would it be possible to toggle between *Share webcam* / *Stop sharing*?
Using the same button for both actions?

<img width="379" height="214" alt="image" src="https://github.com/user-attachments/assets/c13cb2a3-5150-4cc8-aff9-330973813e06" />

----

The carousel feels unintuitive because the items right after it (presence / audio controls) never change, so only those further down shift. This creates confusion. I wouldn’t recommend using a carousel for camera switching, it doesn’t feel natural.

<img width="405" height="769" alt="image" src="https://github.com/user-attachments/assets/1fe98a0e-4ed2-4770-8298-7fda0cc939d2" />

prlanzarin commented:
> Taking a look at the images, I had some thoughts, just sharing in case you consider it
> [...]

This has been forwarded to the design team.

prlanzarin commented:
As per Felix (UI/UX)/Lucas' recommendations, we'll proceed with reviewing + merging this as-is and tackle improvement suggestions in subsequent PRs once more feedback has been aggregated from actual internal testing.

Arthurk12 reviewed (commented):
First of all, very nice work. Having a dedicated hook to handle all the heavy preview logic makes the code easier to reason about and ensures consistency across all client video previews (camera as content, profile tab, and legacy camera modal).  

---

### Bug

- There is one odd behavior that I've found when testing this new feature, where the "Share webcam" button gets active when it shouldn't.
  Steps to reproduce:  
  1. Share one camera.
  2. The "share webcam" button is disabled.  
  3. Click “add extra camera”.  
     - A new camera section is added, but the preview remains on the previously shared camera.  
     - The *share* button becomes available again.  
  4. Click "share webcam".  
     - No camera is actually shared, the preview only switches to the additional camera.  
  5. Click again.  
     - Only then the camera is shared.  

  **Expected behavior:** when adding an extra camera, the preview should automatically switch to the new section to avoid confusion.
  ![bug_share_button](https://github.com/user-attachments/assets/3c084bd3-ea32-4b3a-9bad-6c1ae207fe00)



- MINOR: Sometimes, revoking the browser permisison while preview in profile tab was active, caused an error where the webcam selector got empty causing this error on console.
<details> <summary>Click here to expand console error</summary>

 ```
Warning: `value` prop on `input` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components. Error Component Stack
    at input (<anonymous>)
    at emotion-element-7a1343fa.browser.development.esm.js:48:27
    at SelectInput (SelectInput.js:111:1)
    at emotion-element-7a1343fa.browser.development.esm.js:48:27
    at div (<anonymous>)
    at emotion-element-7a1343fa.browser.development.esm.js:48:27
    at InputBase (InputBase.js:240:32)
    at OutlinedInput (OutlinedInput.js:182:32)
    at emotion-element-7a1343fa.browser.development.esm.js:48:27
    at Select (Select.js:37:32)
    at P (styled-components.browser.esm.js:1:19424)
    at div (<anonymous>)
    at P (styled-components.browser.esm.js:1:19424)
    at div (<anonymous>)
    at P (styled-components.browser.esm.js:1:19424)
    at div (<anonymous>)
    at P (styled-components.browser.esm.js:1:19424)
    at div (<anonymous>)
    at P (styled-components.browser.esm.js:1:19424)
    at ProfileSettings (component.tsx:186:36)
    at div (<anonymous>)
    at P (styled-components.browser.esm.js:1:19424)
    at div (<anonymous>)
    at Resizable (index.js:200:1)
    at div (<anonymous>)
    at P (styled-components.browser.esm.js:1:19424)
    at SidebarContent (component.tsx:28:5)
    at SidebarContentContainer (container.tsx:7:36)
    at div (<anonymous>)
    at P (styled-components.browser.esm.js:1:19424)
    at App (component.tsx:91:3)
    at AppContainer (container.tsx:42:54)
    at DefaultPropsProvider (DefaultPropsProvider.js:9:1)
    at RtlProvider (index.js:6:1)
    at ThemeProvider (ThemeProvider.js:33:1)
    at ThemeProvider (ThemeProvider.js:48:1)
    at ThemeProviderNoVars (ThemeProviderNoVars.js:8:1)
    at ThemeProvider (ThemeProvider.js:9:1)
    at Base (base.jsx:28:22)
    at BaseContainer (base.jsx:108:37)
    at IntlAdapter (intlAdapter.tsx:24:3)
    at PresenceAdapter (component.tsx:10:60)
    at ContextProvider (index.modern.mjs:23:1)
    at LayoutContextProvider (context.jsx:1558:56)
    at CustomBackgroundsProvider (context.jsx:91:45)
    at PluginsContextProvider (context.tsx:9:42)
    at CurrentUserProvider (current-user.tsx:14:69)
    at ContextProvidersComponent (component.jsx:18:10)
    at Startup (meetingClient.jsx:38:40)
    at PresenceManager (component.tsx:60:3)
    at PresenceManagerContainer (component.tsx:226:78)
    at ApolloProvider (ApolloProvider.js:6:1)
    at ConnectionManager (component.tsx:84:64)
    at LoadingScreenHOC (component.tsx:22:3)
    at ErrorBoundary (component.jsx:33:22)
    at CustomStyles (component.tsx:9:11)
    at IntlProvider (provider.js:31:1)
    at IntlLoader (intlLoader.tsx:91:3)
    at IntlLoaderContainer (intlLoader.tsx:184:3)
    at CustomUsersSettings (component.tsx:22:3)
    at SettingsLoader (component.tsx:29:11)
    at Main (<anonymous>)
```
</details>

---

### Behavior Changes (for reference)

#### No video device connected
- Dev31:  
  - Both previews(profile and webcam modal): *“No webcams found. Please make sure there's one connected”*.  
  - Console logs:  
    - Error: `getUserMedia failed in video-preview`  
    - Warning: `Error : no webcams found`  
- PR#23847:  
  - Both previews: same message shown.  
  - Console logs (5 total):  
    - Errors:  
      1. `Error in video getUserMedia: NotFoundError - Requested device not found`  
      2. `getUserMedia failed in video-preview: NotFoundError - Requested device not found`  
      3. `getUserMedia failed in video-preview: Error - NotFoundError`  
    - Warnings:  
      1. `Error displaying final selection`  
      2. `Error : no webcams found`  
- Future improvement: keep a minimum size for the camera preview container when showing error messages. Currently the size depends on the content, so when there is no video stream and only text, the preview area becomes very small.  
<img width="634" height="1284" alt="Screenshot from 2025-09-23 15-23-51" src="https://github.com/user-attachments/assets/f4b99aab-85eb-4478-a8da-0b908aa69433" />

---

### Design Suggestion
This is more of a note, aligned with what Gustavo mentioned and with Paulo’s remark that the design team prefers to merge and iterate later.  

The multi-camera interface is inherently complex. To keep things simple when the user is dealing with only one camera:  
- Use single buttons without mapping per section (for example, one toggle button for *Share* / *Unshare*).  
- Only introduce per-section controls when multiple cameras are actually active.

germanocaumo committed:
fix(html5): profile settings video preview minor fixes

Fixes some issues from review:
- Keep size of preview when showing error messages
- Change preview to the new camera after clicking "Add Extra Camera"
- Only show add extra camera if there is more than one webcam connected
